src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -106,7 +106,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
   <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(HasRuntimeOutput)' == 'true' and '$(RuntimeIdentifier)' != ''">
     <SelfContained Condition="'$(SelfContained)' == ''">true</SelfContained>
-    <UseAppHost Condition="'$(UseAppHost)' == '' and '$(SelfContained)' == 'true'">true</UseAppHost>
+    <UseAppHost Condition="'$(UseAppHost)' == '' and ('$(SelfContained)' == 'true' or '$(_TargetFrameworkVersionWithoutV)' >= '2.2')">true</UseAppHost>
```

Comment:
`'$(HasRuntimeOutput)' == 'true'` is set for test projects, right?  Do we want to use the app host for test projects?

Reply:
That's a good question, and something that probably would need to be addressed in the Test SDK by setting `UseAppHost` to false if desired.

I don't have the background as to why test projects have an exe output type for .NET Core, but generating an apphost for them isn't "bad", per se, since executing it would be the same experience as `dotnet run` (does nothing since `Main` is empty for test assemblies).